### PR TITLE
[CALCITE-5288] '(a > 1 and a < 3) or (a > 2 and a < 4)' can't be simplified

### PR DESCRIPTION
**Pull request by alex-plekhanov**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rex/RexProgramTest.java [hunk 1]
```diff
@@ -1243,6 +1243,13 @@ private void checkExponentialCnf(int n) {
         RelOptPredicateList.of(rexBuilder,
             ImmutableList.of(le(aRef, literal(5)), le(bRef, literal(5)))),
         "false");
+
+    // condition "(a >= 1 and a <= 3) or (a >= 2 and a <= 4)
+    // yelds "a >= 1 and a <= 4"
+    checkSimplifyFilter(
+        or(and(ge(aRef, literal(1)), le(aRef, literal(3))),
+            and(ge(aRef, literal(2)), le(aRef, literal(4)))),
```
Comment (chunweilei) [inline core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1251]: Could you add a test like  "(a >= 1 and a <= 3) or (a >= 0 and a <= 2) -> a>=0 and a<=3 "?

Comment (alex-plekhanov) [reply core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1251]: Done

**Review by libenchao — APPROVED** [review]
+1

**Comment by libenchao** [comment]
I'll merge this PR if there is no objections appear in 24 hours.

**Pull request closed without merging**